### DISCUSSION TITLE
Does English have priority in the hierarchy of languages by architecture?

### DISCUSSION THREAD

**Discussion by d4h0y0sr** [DivinumOfficium/divinum-officium#3868] [Developers corner]
![image](https://github.com/DivinumOfficium/divinum-officium/assets/54406272/f82be71c-dd83-4da8-a467-8f8ac6c843c8)

If i understood well this comment the flow of languages works in this way:

1- Latin
2- English
3- Other languages.

Which mean that if Latin has the correct @ reference, if someone overwrites it in English by mistake, then automatically other languages get in trouble. 

I wish English would be at the same level than other languages, in that way if someone overwrites by mistake a Latin @ reference, the inheritance would not impact other languages.

**Comment by fiapps** [comment]
The reason for the special handling of English is to provide at least some translation of the Latin when there is no translation available in the selected language.

If the Latin section is just a reference, then there should not be any need for the English (or any other language) to provide data for that section: it's sufficient to provide a translation for the referenced text. Things become more complicated when the Latin section is not just a reference, but also adds literal text or makes a regex substitution. In this case, all the languages do need to provide a translation, either with a complete literal text or with a combination similar to that used in Latin.

People who modify the data do make mistakes, and this is why changes need to be tested. Testing is really the key, and the repo contains `admin/divinum-get.pl` and `admin/divinum-replay.pl` for this purpose. You can start a local server with Docker or just do something simple like `cd web; python3 -m http.server --bind localhost --cgi 8000`.

**Reply by d4h0y0sr** [reply]
@fiapps This has been a big issue, i dont see any reason whatsoever to display English instead of Latin when files are missed, we should not take for granted that people would understand english better than latin.

And also, yes, teorically given a reference in Latin section, the vernacular texts should be removed, but this is not the case, this is not documented in the documentation, and its not intuitive, some people even repeat the reference in the subsection. This has been an issue for us, the colaborators of vernaculars, because we have to maintain not only vernacular but also English, if someone messes with English all vernaculars are affected, its a very weak point.

It will be at least better if the algorithm check first if there is a reference in the latin section before choosing to display the english when the text is missing, this would save a lot of work.

**Reply by fiapps** [reply]
@d4h0y0sr it's not up to me to change the policy, I was just explaining how it does work. Since English is the [most widely spoken language in the world](https://www.ethnologue.com/insights/most-spoken-language/), there is nothing users are more likely to understand. Furthermore, falling back to a language other than Latin gives users two options: Latin and the fallback language. If the user knows Latin better than English, he has the Latin to read. If he knows English better, he can read that.

The unnecessary overrides in English are a problem, but one that should eventually be resolved. It ought to be possible to write a script to see where English is overriding a Latin section that consists only of a reference.
